lib.php

```diff
@@ -330,49 +332,27 @@ function learningmap_reset_userdata($data) {
  * @return void
  */
 function learningmap_before_http_headers() {
-    global $PAGE, $DB, $OUTPUT;
+    global $PAGE, $OUTPUT;
 
     if ($PAGE->context->contextlevel != CONTEXT_MODULE) {
         return '';
     }
 
     try {
         $cache = cache::make('mod_learningmap', 'backlinks');
-        $cachekey = $PAGE->cm->id;
-        $key = $cache->get($cachekey);
-        $modinfo = get_fast_modinfo($PAGE->course);
-
-        if (!$key) {
-            $instances = $modinfo->get_instances_of('learningmap');
-            if (count($instances) > 0) {
-                $backlinks = [];
-                foreach ($instances as $i) {
-                    $record = $DB->get_record('learningmap', ['id' => $i->instance], 'name, placestore, backlink');
-                    if ($record->backlink == 1) {
-                        $placestore = json_decode($record->placestore);
-                        $coursepageurl = course_get_format($PAGE->course->id)->get_view_url($i->sectionnum);
-                        $coursepageurl->set_anchor('module-' . $i->id);
-                        foreach ($placestore->places as $place) {
-                            $url = !empty($i->showdescription) ?
-                                $coursepageurl->out() :
-                                new moodle_url('/mod/learningmap/view.php', ['id' => $i->id]);
-                            $backlinks[$place->linkedActivity][$i->id] = ['url' => $url, 'name' => $record->name, 'cmid' => $i->id];
-                        }
-                    }
-                }
-                foreach ($backlinks as $cmid => $backlink) {
-                    $cache->set($cmid, $backlink);
-                }
-            } else {
-                $cache->set($cachekey, []);
-            }
-        } else {
-            $backlinks[$cachekey] = $key;
+
+        if (!$cache->get('fillstate')) {
```

Comment:
Please comment that this is a real edge case, because this will only happen when someone purged the caches and the task has not run yet to rebuild the whole cache. While writing ... do we really need this? The task currently checks every minute if the caches are valid. Purging the caches is a really rare operation. Don't we just want to rely on the fact that the task takes care of everything?

Reply:
I'd like to keep that option for the case that cache filling needs more time than expected (so backlink function won't become unreliable).